### ISSUE TITLE
Typization issue with `defaultComponent` in react-native project

### ISSUE THREAD

**Issue by aymkin** [lingui/js-lingui#2066]
**Describe the bug**
I try to use prop `defaultComponent` as described in Guide here https://lingui.dev/tutorials/react-native#internationalization-in-react-native

<img width="746" alt="image" src="https://github.com/user-attachments/assets/481b2947-8f4a-4f71-beb8-220143b30b02">

**To Reproduce**
- Not sure if that is reproducable
- init expo app 
- eject it
- install lingui
- try to use `defaultComponent` in `I18nProvider`
- get the typization error 
```shell
TS2322: Type typeof Text is not assignable to type ComponentType<TransRenderProps> | undefined
Type typeof Text is not assignable to type ComponentClass<TransRenderProps, any>
Construct signature return types Text and Component<TransRenderProps, any, any> are incompatible.
The types of props are incompatible between these types.
Type Readonly<TextProps> is missing the following properties from type Readonly<TransRenderProps>: translation, isTranslated```

**Expected behavior**
Should not be any problem

**Additional context**
Add any other context about the problem here.

- jsLingui version 4.13.0
- Babel version 
```
├─┬ @babel/core@7.25.9
│ └─┬ @babel/helper-module-transforms@7.25.9
│   └── @babel/core@7.25.9 deduped
├─┬ @lingui/cli@4.13.0
│ └── @babel/core@7.25.9 deduped
├─┬ @react-native/metro-config@0.75.4

```
- Macro support:
 - [NO] I'm using SWC with `@lingui/swc-plugin`
 - [NO] I'm using Babel with `babel-macro-plugin`
- Your Babel config (e.g. `.babelrc`) or framework you use (Create React App, NextJs, Vite)
```
module.exports = function (api) {
  api.cache(true);
  return {
    presets: ["babel-preset-expo"],
    plugins: ["macros"],
  };
};
```

**Comment by andrii-bodnar** [comment]
It looks like the referenced example in the RN tutorial is wrong.

Please refer to the example project:

https://github.com/lingui/js-lingui/blob/e726b16a2a9299913d386489d92e0daa9c42e40d/examples/react-native/App.tsx#L18-L26

In addition, the tests may shed some light on this:

https://github.com/lingui/js-lingui/blob/e726b16a2a9299913d386489d92e0daa9c42e40d/packages/react/src/Trans.test.tsx#L336-L348

**Comment by ashuvssut** [comment]
Hi! I’m new to Lingui -- thank you for all the work on the project

Could you please help me understand why the `defaultComponent` prop might be required?
Is it intended to act as a fallback component (similar to a loader), or does it serve a different purpose?

**Comment by timofei-iatsenko** [comment]
@ashuvssut i'm not an expert in ReactNative, but i believe that in RN you could not render text without using `<Text>` component. So that's why you have to pass it so lingui's Trans whould render the translation wrapped in `<Text>`. 

If i'm not mistaken you can achieve the same by changing the composition to:

```tsx
<Text><Trans>Hello!</Trans</Text>
```

But i haven't tried that myself.

**Comment by ashuvssut** [comment]
Hi @timofei-iatsenko! thanks for replying. Sorry for the confusion -- that wasn’t quite what I meant. let me rephrase my question again with an example:-

Lets say we have this JSX

```tsx
const DefaultComponent = (props: TransRenderProps) => {
  return <Text>{props.children}</Text>;
};

export default function Root() {
  return (
    <I18nProvider
      // defaultComponent={DefaultComponent} // commented out
      i18n={i18n} 
    >
      <MyEntireApp />
    </I18nProvider>
  );
}

```

in this setup, the app works perfectly fine just by wrapping `<MyEntireApp />` with `I18nProvider`.
why might we need to put a `defaultComponent`? when does it render? I mean, I want to understnad **what role does `defaultComponent` play here**

I noticed that in Bluesky’s implementation of `I18nProvider` code, they didnt pass the `defaultComponent` prop at all.
- Checkout this snippet from blusky-social repo
	
	https://github.com/bluesky-social/social-app/blob/3be1f5caa9095bc9348fd770cfe5cacc7c9c696f/src/locale/i18nProvider.tsx#L7-L10

 I understand that `defaultComponent` prop is optional. But I wanted to understand its use case.

**Comment by ashuvssut** [comment]
Oh, maybe I understand it now.

When using the `t` function directly in JSX like this:

```tsx
const HomePage = () => {
  return (
    <View>
      {t`Hello world`}
    </View>
  );
};
```

Lingui will automatically render it as:

```tsx
<View>
  <Text>Hello world</Text>
</View>
```

And when using the `t` function inside a `<Text>` component:

```tsx
const HomePage = () => {
  return (
    <View>
      <Text>
        {t`Hello world`}
      </Text>
    </View>
  );
};
```

Lingui will render it as:

```tsx
<View>
  <Text>
    <Text>Hello world</Text>
  </Text>
</View>
```

thanks.
